### PR TITLE
[Core] Improve StringUtilities

### PR DESCRIPTION
**Pull request by loumalouomega**
Just extend StringUtilities

### CODE REVIEW

File: kratos/tests/cpp_tests/utilities/test_string_utilities.cpp [hunk 1]
```diff
@@ -25,5 +25,19 @@ KRATOS_TEST_CASE_IN_SUITE(ConvertCammelCaseToSnakeCase, KratosCoreFastSuite)
     KRATOS_CHECK_STRING_EQUAL(snake_case, "test_in_cammel_case");
 }
 
+KRATOS_TEST_CASE_IN_SUITE(ErasePartialString, KratosCoreFastSuite)
+{
+    const std::string text_with_stuff = "textwithstuff";
```
Comment (philbucher) [inline kratos/tests/cpp_tests/utilities/test_string_utilities.cpp:34]: can you please add a test with erasing "with"

File: kratos/tests/cpp_tests/utilities/test_string_utilities.cpp [hunk 2]
```diff
@@ -25,5 +25,19 @@ KRATOS_TEST_CASE_IN_SUITE(ConvertCammelCaseToSnakeCase, KratosCoreFastSuite)
...
+    const std::string text = StringUtilities::ErasePartialString(text_with_stuff, "withstuff");
+    KRATOS_CHECK_STRING_EQUAL(text, "text");
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RemoveWhiteSpaces, KratosCoreFastSuite)
+{
+    const std::string text_with_spaces = "textwithspaces                ";
```
Comment (philbucher) [inline kratos/tests/cpp_tests/utilities/test_string_utilities.cpp]: please add also a test removing whitespaces in the middle of a string

File: kratos/utilities/string_utilities.cpp [hunk 3]
```diff
@@ -14,6 +14,7 @@
 #include <algorithm>
 #include <iostream>
 #include <cctype>
+#include <string>
```
Comment (philbucher) [inline kratos/utilities/string_utilities.cpp]: this needs to be in the header

**Review by philbucher — COMMENTED** [review]
minor otherwise ok for me

**Comment by loumalouomega** [comment]
@philbucher done

**Review by philbucher — APPROVED** [review]

**Pull request merged**